iptables/iptables.go

```diff
@@ -94,8 +93,19 @@ func NewChain(name, bridge string, table Table, hairpinMode bool) (*Chain, error
 			return nil, fmt.Errorf("Could not create %s/%s chain: %s", c.Table, c.Name, output)
 		}
 	}
+	return c, nil
+}
 
-	switch table {
+// ProgramChain is used to add rules to a chain
+func ProgramChain(c *ChainInfo, hairpinMode bool) (*ChainInfo, error) {
+	if c.Name == "" {
+		return c, fmt.Errorf("Could not program chain, missing chain name.")
+	}
+	if string(c.Table) == "" {
```

Comment:
This seems redundant. This is already done in `NewChain` so it does need to be setup here as well